### PR TITLE
KI-15396: Device Pin Fallback

### PR DESCRIPTION
**Pull request by VladimirGradev**
This PR introduces an additional param to the native plugin code `authenticationDevicePinFallback` which is used for Device PIN fallback upon biometric challenge.

Example usage can be found [here](https://github.com/tideplatform/universal-client/pull/9765)

### CODE REVIEW

File: macos/Classes/BiometricStorageImpl.swift [hunk 1]
```diff
@@ -289,7 +281,8 @@ class BiometricStorageFile {
       kSecValueData as String: content.data(using: String.Encoding.utf8) as Any,
     ]) { (_, new) in new }
     var status = SecItemAdd(query as CFDictionary, nil)
-    if (status == errSecDuplicateItem) {
+    
+    if (status == errSecDuplicateItem || initOptions.authenticationDevicePinFallback) {
```
Comment (benjipetit) [inline macos/Classes/BiometricStorageImpl.swift:285]: these 2 seem unrelated. Did you test the case where we don't have a value and we use pin fallback?
Wouldn't `query.removeValue` crash?

Comment (VladimirGradev) [reply macos/Classes/BiometricStorageImpl.swift:285]: This change is a cosmetic one. Apple don't show the biometric challenge on SecItemAdd (It is shown only if there is a value and it is doing an update). QA dept however needs to have it always in the recovery flow, that's why I added the update where it is shown. 
When we move 100% to keyAuth this will be removed.
I tested both having and not having a value.
